src/Object/Contact.php

```diff
@@ -99,18 +100,12 @@ public static function markForArchival(array $contact)
 		}
 
 		if ($contact['term-date'] <= NULL_DATE) {
-			q(
-				"UPDATE `contact` SET `term-date` = '%s' WHERE `id` = %d", dbesc(datetime_convert()), intval($contact['id'])
-			);
+			dba::update('contact', array('term-date' => datetime_convert()), array('id' => $contact['id']));
 
 			if ($contact['url'] != '') {
-				q(
-					"UPDATE `contact` SET `term-date` = '%s'
-				WHERE `nurl` = '%s' AND `term-date` <= '1000-00-00'", dbesc(datetime_convert()), dbesc(normalise_link($contact['url']))
-				);
+				dba::update('contact', array('term-date' => datetime_convert()), array('`nurl` = ? AND `term-date` <= 1000-00-00', normalise_link($contact['url'])));
```

Comment:
You haven't put the "1000-00-00" in quotes. But possibly you should replace it with ```NULL_DATE``` (Yeah, I know, I should decide) :-)